javascript/javascript3/week3/shopping-cart/main.js

```diff
@@ -0,0 +1,118 @@
+class Product {
+  constructor(name, price) {
+    this.name = name;
+    this.price = price;
+  }
+
+  convertToCurrency(currency) {
+    fetch(`https://api.exchangerate-api.com/v4/latest/DKK`)
+      .then((response) => response.json())
+      .then((data) => {
+        let currencyConvert = (data.rates[currency] * this.price).toFixed(2);
+        console.log(
+          `Price of ${this.name} in ${currency} is ${currencyConvert}`
+        );
+      });
+  }
+}
+
+class ShoppingCart {
+  constructor() {
+    this.products = [];
+  }
+
+  addProduct(product) {
+    // Implement functionality here
+    this.products.push(product);
+  }
+
+  removeProduct(product) {
+    // Implement functionality here
+    const removedProduct = this.products.filter(
+      (itemname) => product !== itemname
+    );
+    this.products = removedProduct;
+  }
+
+  searchProduct(productName) {
+    // Implement functionality here
+    let SearchProductName = this.products.filter(
+      (itemname) => itemname === productName
+    );
+    console.log(SearchProductName);
+  }
+
+  getTotal() {
+    // Implement functionality here
+    const pricelist = this.products.map((product) => product.price);
+    const TotalPrice = pricelist.reduce((sum, item) => sum + item);
+    console.log(TotalPrice);
+    return TotalPrice;
+  }
+
+  renderProducts() {
+    const container = document.querySelector("#products");
+    const allnames = this.products.map(
+      (item) => `${item.name}    --   ${item.price} Kr`
+    );
+
+    allnames.forEach((sub) => {
+      const item = document.createElement("li");
+      item.innerHTML = sub;
+      container.appendChild(item);
+    });
+  }
+
+  getUser() {
+    // Implement functionality here
+    fetch(`https://jsonplaceholder.typicode.com/users/1`)
+      .then((Response) => Response.json())
+      // .then(data=> console.log(data))
+      .then((user) => {
+        this.renderuser(user);
```

Comment:
i would not call `renderUser` from here, but from where you're calling getUser (see my other comment). 
Why? Usually you want to separate data operations (numbers, text), from DOM operations (createElement, appendChild, innerText, etc)